### ISSUE TITLE
Cannot create library, cannot create user  "Verify Token User Not Found"

### ISSUE THREAD

**Issue by TheNomad11** [YunoHost-Apps/audiobookshelf_ynh#20]
### Describe the bug

After installing the app and logging in as default root user without password I cannot create a new user, I cannot create a library either. I get error messages instead. "Library must have a name" (when creating a library with a name) or "Not found" (when creating a user)

Looking at the logs I find the following:

`ERROR: Verify Token User Not Found bG9yZW56Oitib2hlTTE1eQ==`

or

`ERROR: JWT Verify Token Failed JsonWebTokenError: jwt malformed`

Full log: https://paste.yunohost.org/zokajewiyi

### Context

- Hardware: *VPS bought online.*
- YunoHost version: 4.3.6.3
- I have access to my server: *Through SSH | through the webadmin.*
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *no*

**Comment by navanchauhan** [comment]
Getting a "Not Found" error in both cases:
* Creating a new user
* Creating a new library

**Comment by navanchauhan** [comment]
Might be related to https://github.com/advplyr/audiobookshelf/commit/c962090c3aa617b438a0dd06854a9eb012272568

**Comment by navanchauhan** [comment]
Ummmmmm, so the solution to fix this problem is to first remove the app completely `sudo yunohost app remove --purge audiobookshelf` 

So, basically what happened is that now there is no default user `root` with a blank password, but you are asked to create the user you open the page for the first time. What happened in my test system was that I still had the old (cache/database?) in my system from a previous install before this was implemented, so the app got confused. 

Purging everything and then installing using the web admin panel or `sudo yunohost app install audiobookshef` will fix it

**Comment by Qprq** [comment]
Same issue here (fresh install today). Disconnecting from SSOWat solved the issue.

**Comment by navanchauhan** [comment]
> Same issue here (fresh install today). Disconnecting from SSOWat solved the issue.

By disconnecting, do you mean editing the nginx configuration and removing the SSO panel or by logging out and then going to the Audiobookshelf URL?

**Comment by navanchauhan** [comment]
> Thanks so much for working with this issue. Does it work for you? I purged it and reinstalled but do get the same error messages as before. What are the correct steps after starting the app? Creating a root user root without a password? Or another root user with a different name and password? Or something else?

With the latest update you don't need to create a root user with the username `root`, you can use any username and I would recommend to not use a blank password

**Comment by navanchauhan** [comment]
Please reopen if this still persists

**Comment by majamee** [comment]
I have the same issue with a fresh install from today. Also tried to purge and reinstall, but this does not change the behavior.

I would also like to know what this means "Disconnecting from SSOWat" : where/ how to do/ try that?

**Comment by czen** [comment]
Reproduced the issue today on yunohost 4.3.6.3. 
If you install Audiobookshelf package with no SSOwat panel in  Nginx config, allow public access and make sure no previous app settings remain in /home/yunohost.app folder, the app installs just fine.
After that you have to log out of yunohost in the browser. If you don't log out, every request to Audiobookshelf API ends with 404 error. At least if you try to create the first library or add users immediately after installing the app.

**Comment by 1appuser** [comment]
I have the same issue on a fresh install. The proposed solution for purging and reinstalling did not work for me. Is there anything else that can be tried for this issue? I see others are still experiencing it as well.

**Comment by navanchauhan** [comment]
I always recommend trying to login via private/incognito mode to test.

Otherwise, I’m on a winter break from school now. Give me a few days to
take a break, and then I’ll look into this again

Best,
Navan

**Comment by 1appuser** [comment]
Thank you. I already tried multiple browsers, including private/incognito modes on all of them, but no luck. Thanks for looking into it, and enjoy your winter break. Happy Holidays. :)

**Comment by bauerbyter** [comment]
Sorry have to reopen this.
I installed it yesterday and have the same error. Even with private mode browser. What can I do?

**Comment by stoltenhoff** [comment]
… tried it today, still no luck.

**Comment by RedGlow** [comment]
I've had the same problem. As far as I've been able to debug, the problem was that my browser was sending the correct Authorization header to the application (Bearer + JWT token), whereas the application was receiving a wrong one (Basic authorization header with the yunohost credentials (?!?!)).

I logged out of the admin panel, cleared all cookies and everything, and at that point I was able to log in. So, in my case at least, it seems like some sort of conflict between yunohost panel and the app? But I have not even an idea how and who changes the Authorization header in this situation.
